Iceberg-TipUI/IceTipSelectAllPackagesCommand.class.st

```diff
@@ -0,0 +1,50 @@
+"
+Implements the select all option behavior in the ""Select New Packages to Load"" Iceberg dialog.
+
+"
+Class {
+	#name : #IceTipSelectAllPackagesCommand,
+	#superclass : #IceTipSelectPackagesCommand,
+	#category : #'Iceberg-TipUI-Commands'
+}
+
+{ #category : #default }
+IceTipSelectAllPackagesCommand class >> defaultDescription [
+
+	^ 'Opens a dialog to choose select and/or removing new package(s) from the repository'
+]
+
+{ #category : #default }
+IceTipSelectAllPackagesCommand class >> defaultName [
+
+	^ 'Select all packages'
+]
+
+{ #category : #testing }
+IceTipSelectAllPackagesCommand >> canBeExecuted [
+	" Answer <true> if the receiver has items to display and they are not all selected "
+
+	| pkgList |
+
+	pkgList := self context packagesList.
+ 	^ pkgList items notEmpty and: [ pkgList items size ~= pkgList selectedItems size ]
+]
+
+{ #category : #executing }
+IceTipSelectAllPackagesCommand >> execute [
+	" Private - Select all the receiver's packages "
+
+	self packagesList items
+		do: [ : pkgItem |
+			self packagesListSelectionColumn
+				select: [ : c | c isKindOf: SpCheckBoxTableColumn ]
+				thenDo: [ : c | c onActivation cull: pkgItem ] ]
+		displayingProgress: [ : pkgItem | 'Selecting ' , pkgItem name ].
```

Comment:
This method looks fishy:
 - is `c isKindOf: SpCheckBoxTableColumn` the only way to select all checkboxes in a colum in a table? Maybe @estebanlm knows better, I cannot help here
 - Why a progress bar here? How many packages do we need in a repository so that this is SOOO slow that the progress bar becomes actually visible and gives reasonable feedback? :)

Reply:
> * Why a progress bar here? How many packages do we need in a repository so that this is SOOO slow that the progress bar becomes actually visible and gives reasonable feedback? :)

IMO selection is slow because of the current implementation, which is the best I could arrive with the current Spec 2 API.  So we should measure here, but if there's a better way to select items I'm looking forward to change it.

Reply:
Let's wait for @estebanlm's input.
Maybe it's slow because it's redrawing the entire table for each selection?
Maybe we need a new API to select many things at the same time?